pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -862,7 +862,7 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 		// is done and there is no container to start.
 		if len(containersToStart) == 0 {
 			hasInitialized := false
-			if !utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+			if !types.HasRestartableInitContainer(pod) {
```

Comment:
compute once at the top of computePodActions:
```go
handleInitContainerSidecars := utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) && types.HasRestartableInitContainer(pod)
```

then use that throughout `computePodActions`